tests/testImageMapReduce.py

```diff
@@ -130,58 +142,122 @@ class ImageMapReduceTest(lsst.utils.tests.TestCase):
+        self._testCopySumNoOverlaps(reduceOp='sum', withNaNs=False)
+        self._testCopySumNoOverlaps(reduceOp='sum', withNaNs=True)
 
-    def testExampleTaskNoOverlaps(self):
+    def _testCopySumNoOverlaps(self, reduceOp='copy', withNaNs=False):
         """Test sample grid task that adds 5.0 to input image and uses
-        default 'copy' `reduceOperation`.
+        default 'copy' `reduceOperation`. Optionally add NaNs to subimages.
         """
         config = AddAmountImageMapReduceConfig()
         task = ImageMapReduceTask(config)
         config.mapperSubtask.addAmount = 5.
-        newExp = task.run(self.exposure)
+        config.reducerSubtask.reduceOperation = reduceOp
+        newExp = task.run(self.exposure, addNans=withNaNs).exposure
         newMI = newExp.getMaskedImage()
         newArr = newMI.getImage().getArray()
-        mi = self.exposure.getMaskedImage()
+        isnan = np.isnan(newArr)
+        if not withNaNs:
+            self.assertTrue(np.sum(isnan) == 0,
+                            msg='Failed on withNaNs: %s' % str(withNaNs))
+
+        mi = self.exposure.getMaskedImage().getImage().getArray()
+        if reduceOp != 'sum':
+            self.assertFloatsAlmostEqual(mi[~isnan], newArr[~isnan] - 5.,
+                                         msg='Failed on withNaNs: %s' % str(withNaNs))
 
-        self.assertFloatsAlmostEqual(mi.getImage().getArray(), newArr - 5.)
+    def testAverageWithOverlaps(self):
+        self._testAverageWithOverlaps(withNaNs=False)
+        self._testAverageWithOverlaps(withNaNs=True)
 
-    def testExampleTaskWithOverlaps(self):
+    def _testAverageWithOverlaps(self, withNaNs=False):
         """Test sample grid task that adds 5.0 to input image and uses
-        'average' `reduceOperation`.
+        'average' `reduceOperation`. Optionally add NaNs to subimages.
         """
         config = AddAmountImageMapReduceConfig()
         config.gridStepX = config.gridStepY = 8
         config.reducerSubtask.reduceOperation = 'average'
         task = ImageMapReduceTask(config)
         config.mapperSubtask.addAmount = 5.
-        newExp = task.run(self.exposure)
+        newExp = task.run(self.exposure, addNans=withNaNs).exposure
         newMI = newExp.getMaskedImage()
         newArr = newMI.getImage().getArray()
         mi = self.exposure.getMaskedImage()
+        isnan = np.isnan(newArr)
+        if not withNaNs:
+            self.assertTrue(np.sum(isnan) == 0,
+                            msg='Failed on withNaNs: %s' % str(withNaNs))
+
+        mi = self.exposure.getMaskedImage().getImage().getArray()
+        self.assertFloatsAlmostEqual(mi[~isnan], newArr[~isnan] - 5.,
+                                     msg='Failed on withNaNs: %s' % str(withNaNs))
+
+    def _testAverageVersusCopy(self):
+        self.testAverageVersusCopy(withNaNs=False)
+        self.testAverageVersusCopy(withNaNs=True)
+
+    def testAverageVersusCopy(self, withNaNs=False):
```

Comment:
This should have a preceeding `_`, otherwise it'll get run by the test suite on its own.